main.py

```diff
@@ -180,6 +180,258 @@ def canonical_idx4_reverse(ijkl):
+    assert da == db
+    assert (i in da.alpha) and (i in da.beta)
+    return
+
+
+def test_pair_idx_B(dadb, idx):
+    da, db = dadb
+    assert integral_category(*idx) == "B"
+    i, j, _, _ = idx
+    assert (i, j, i, j) == idx
+    assert da == db
+    assert (i in da.alpha + da.beta) and (j in da.alpha + da.beta)
+    return
+
+
+def test_pair_idx_C(dadb, idx):
+    da, db = dadb
+    assert integral_category(*idx) == "C"
+    i, j, k, l = idx
+    exc = Excitation.exc_degree(da, db)
+    if exc == (1, 0):
+        dsa = da.alpha
+        dsb = db.alpha
+    elif exc == (0, 1):
+        dsa = da.beta
+        dsb = db.beta
+    else:
+        raise
+    _, h, p = PhaseIdx.single_exc(dsa, dsb)
+    if j == l:
+        assert sorted((h, p)) == sorted((i, k))
+        assert (j in da.alpha + da.beta) and (j in db.alpha + db.beta)
+    elif i == k:
+        assert sorted((h, p)) == sorted((j, l))
+        assert (i in da.alpha + da.beta) and (i in db.alpha + db.beta)
+    else:
+        raise
+    return
+
+
+def test_pair_idx_D(dadb, idx):
+    da, db = dadb
+    assert integral_category(*idx) == "D"
+    i, j, k, l = idx
+    exc = Excitation.exc_degree(da, db)
+    if exc == (1, 0):
+        dsa = da.alpha
+        dsb = db.alpha
+        dta = da.beta
+        dtb = db.beta
+    elif exc == (0, 1):
+        dsa = da.beta
+        dsb = db.beta
+        dta = da.alpha
+        dtb = db.alpha
+    else:
+        raise
+    _, h, p = PhaseIdx.single_exc(dsa, dsb)
+    if j == l:
+        assert sorted((h, p)) == sorted((i, k))
+        assert (j in dta) and (j in dtb)
+    elif i == k:
+        assert sorted((h, p)) == sorted((j, l))
+        assert (i in dta) and (i in dtb)
+    else:
+        raise
+    return
+
+
+def test_pair_idx_E(dadb, idx):
+    da, db = dadb
+    assert integral_category(*idx) == "E"
+    i, j, k, l = idx
+    exc = Excitation.exc_degree(da, db)
+    if sum(exc) == 1:
+        if exc == (1, 0):
+            dsa = da.alpha
+            dsb = db.alpha
+        elif exc == (0, 1):
+            dsa = da.beta
+            dsb = db.beta
+        else:
+            raise
+        _, h, p = PhaseIdx.single_exc(dsa, dsb)
+        if i == j:
+            assert sorted((h, p)) == sorted((k, l))
+            assert (i in dsa) and (i in dsb)
+        elif j == k:
+            assert sorted((h, p)) == sorted((i, l))
+            assert (j in dsa) and (j in dsb)
+        elif k == l:
+            assert sorted((h, p)) == sorted((i, j))
+            assert (k in dsa) and (k in dsb)
+        else:
+            raise
+    elif sum(exc) == 2:
+        if exc == (1, 1):
+            dsa, dsb = da.alpha, db.alpha
+            dta, dtb = da.beta, db.beta
+        else:
+            raise
+        if i == j:
+            p, r, s = i, k, l
+        elif j == k:
+            p, r, s = j, i, l
+        elif k == l:
+            p, r, s = k, j, i
+        else:
+            raise
+        _, hs, ps = PhaseIdx.single_exc(dsa, dsb)
```

Comment:
We should put a new argumetn in `single_exc` to not ask for the phase
```suggestion
        hs, ps = PhaseIdx.single_exc(dsa, dsb,phase_not_required=True)
```

Reply:
I added a separate function (didn't want to deal with input-dependent type hinting or overloading, but we can change it if you'd prefer to do it a different way)